abn/metricstore/summary_metric.go

```diff
@@ -0,0 +1,95 @@
+package metricstore
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/iter8-tools/iter8/base/log"
+)
+
+type SummaryMetric []float64
+
+const (
+	COUNT_IDX = 0
+	SUM_IDX   = 1
+	MIN_IDX   = 2
+	MAX_IDX   = 3
+	SS_IDX    = 4
+	TS_IDX    = 5
+)
+
+func (m *SummaryMetric) Count() uint32 {
+	return uint32(math.Round((*m)[COUNT_IDX]))
+}
+
+func (m *SummaryMetric) SetCount(v uint32) {
+	(*m)[COUNT_IDX] = float64(v)
+}
+
+func (m *SummaryMetric) Sum() float64 {
+	return (*m)[SUM_IDX]
+}
+
+func (m *SummaryMetric) SetSum(v float64) {
+	(*m)[SUM_IDX] = v
+}
+
+func (m *SummaryMetric) Min() float64 {
+	return (*m)[MIN_IDX]
+}
+
+func (m *SummaryMetric) SetMin(v float64) {
+	if v < m.Min() {
+		(*m)[MIN_IDX] = v
+	}
+}
+
+func (m *SummaryMetric) Max() float64 {
+	return (*m)[MAX_IDX]
+}
+
+func (m *SummaryMetric) SetMax(v float64) {
+	if v > m.Max() {
+		(*m)[MAX_IDX] = v
+	}
+}
+
+func (m *SummaryMetric) SumSquares() float64 {
+	return (*m)[SS_IDX]
+}
+
+func (m *SummaryMetric) SetSumSquares(v float64) {
+	(*m)[SS_IDX] = v
+}
+
+func (m *SummaryMetric) LastUpdateTimestamp() time.Time {
+	return time.Unix(int64(math.Round((*m)[TS_IDX])), 0)
+}
+
+func (m *SummaryMetric) SetLastUpdateTimestamp(t time.Time) {
+	(*m)[TS_IDX] = float64(t.Unix())
+}
```

Comment:
Why is it necessary to cast the Unix time stamp from `float64` back into `int64`?

Reply:
`math.Round` returns a `float64`; `time.Unix()` wants an `int64`. Hence the conversion.